src/streamlink/plugins/cnews.py

```diff
@@ -1,25 +1,43 @@
 import re
 
 from streamlink.plugin import Plugin
-from streamlink.plugin.api import useragents
+from streamlink.plugin.api import validate
+from streamlink.utils import parse_json
 
 
 class CNEWS(Plugin):
     _url_re = re.compile(r'https?://www.cnews.fr/[^ ]+')
-    _embed_video_url_re = re.compile(r'class="dm-video-embed_video" src="(?P<dm_url>.*)"')
-    _embed_live_url_re = re.compile(r'class="wrapper-live-player main-live-player"><iframe src="(?P<dm_url>.*)"')
+    _json_data_re = re.compile(r'({"basePath":.*})\);')
+    _dailymotion_url = 'https://www.dailymotion.com/embed/video/{}'
+
+    _data_schema = validate.Schema(
+        validate.any(None, validate.all(
+            validate.transform(_json_data_re.search),
+            validate.any(None, validate.all(
+                validate.get(1),
+                validate.transform(parse_json),
+                validate.filter(lambda k, v: 'dm_player_' in k),
+                {
+                    validate.text: {
+                        'video_id': validate.text,
+                    },
+                },
```

Comment:
Don't filter by "substring in key". Just add `validate.optional()` keys for the two properties to the dict validation. Btw, `validate.text` can be replaced with `str` now.